### DISCUSSION TITLE
Object detection: : `'MultirotorClient' object has no attribute 'simSetDetectionFilterRadius'`

### DISCUSSION THREAD

**Discussion by dzungdoan6** [microsoft/AirSim#3871] [Support Q&A]
Hi everyone, 
I am trying to get the ground truth bounding boxes for training my Faster RCNN. I am using this below code to obtain bounding boxes for cars but Airsim (v1.5.0) reports this error: `AttributeError: 'MultirotorClient' object has no attribute 'simSetDetectionFilterRadius'`. Could you please help me which Airsim version has this feature? Thank you very much!
Here is my complete code:

```
import airsim
import cv2

def main():
    drone_name = "Drone1";

    # connect to the AirSim simulator
    client = airsim.MultirotorClient()
    client.confirmConnection()
   
    # set camera name and image type to request images and detections
    camera_name = "0"
    image_type = airsim.ImageType.Scene

    # set detection radius in [cm]
    client.simSetDetectionFilterRadius(camera_name, image_type, 200 * 100, drone_name)
    # add desired object name to detect in wild card/regex format
    client.simAddDetectionFilterMeshName(camera_name, image_type, "Car_*", drone_name);

    while True:
        rawImage = client.simGetImage(camera_name, image_type)
        if not rawImage:
            continue
        png = cv2.imdecode(airsim.string_to_uint8_array(rawImage), cv2.IMREAD_UNCHANGED)
        cars = client.simGetDetections(camera_name, image_type)
        if cars:
            for car in cars:
                # s = pprint.pformat(car)
                # print("Car: %s" % s)

                cv2.rectangle(png, (int(car.box2D.min.x_val), int(car.box2D.min.y_val)),
                              (int(car.box2D.max.x_val), int(car.box2D.max.y_val)), (255, 0, 0), 2)
                cv2.putText(png, car.name, (int(car.box2D.min.x_val), int(car.box2D.min.y_val - 10)),
                            cv2.FONT_HERSHEY_SIMPLEX, 0.5, (36, 255, 12))

        cv2.imshow("AirSim", png)
        if cv2.waitKey(1) & 0xFF == ord('q'):
            break
        elif cv2.waitKey(1) & 0xFF == ord('c'):
            client.simClearDetectionMeshNames(camera_name, image_type)
        elif cv2.waitKey(1) & 0xFF == ord('a'):
            client.simAddDetectionFilterMeshName(camera_name, image_type, "Car_*")
    cv2.destroyAllWindows()

if __name__ == "__main__":
    main();
```

**Comment by alonfaraj** [comment]
Hi @dadung

You should add `import setup_path` at the top of your script and make sure you have `setup_path.py` in the same folder where you running your script from.
You can see it in the [example script ](https://github.com/microsoft/AirSim/blob/master/PythonClient/detection/detection.py)

**Reply by dzungdoan6** [reply]
thanks @alonfaraj  for your reply. 
I did, following your suggestion, now I am getting this error:
`msgpackrpc.error.RPCError: rpclib: server could not find function 'simAddDetectionFilterMeshName' with argument count 4.`

**Reply by dzungdoan6** [reply]
Hmm, I am using `AirSimNH.zip` downloaded from the [Releases (v1.5.0)](https://github.com/microsoft/AirSim/releases). Is it because `AirSimNH.zip` doesn't have the function `simAddDetectionFilterMeshName`?

**Reply by alonfaraj** [reply]
Yes, this feature was added the official version released. 
You can use it if you compile airsim from source, until the next release
